gradefetcher/gradefetcher.py

```diff
@@ -302,113 +321,138 @@ def grade_user(self, data, suffix=""):
+
+                        total_grade = 0
+                        for g in grades:
+                            total_grade += g
+                        grade = int(truediv(total_grade * 100, len(grades)))
+                    else:
+                        grade = grades[0] * 100
+                    reasons = []
+                    for result in grader_response.json()["results"]:
+                        if "grade" in result:
+                            if result["grade"] > 0:
+                                reason = self.i18n_service.gettext(
+                                    "Assignment {assignment_id}: <b>Passed</b>"
+                                ).format(
+                                    assignment_id=result["assignment_id"],
+                                )
+                                reasons.append(reason)
+                            elif result["grade"] == 0:
+                                reason_api_text = self.i18n_service.gettext(
+                                    result["reason"]
+                                )
+                                reason = self.i18n_service.gettext(
+                                    "Assignment {id}: <b>Failed</b> - {reason}"
+                                ).format(
+                                    id=result["assignment_id"],
+                                    reason=reason_api_text,
+                                )
+                                reasons.append(reason)
+                        elif "grade" not in result:
                             reason_api_text = self.i18n_service.gettext(
                                 result["reason"]
-                            )
+                                )
                             reason = self.i18n_service.gettext(
-                                "Assignment {id}: <b>Failed</b> - {reason}"
+                                "Assignment {assignment_id}: {reason_api_text}"
                             ).format(
-                                id=result["assignment_id"],
-                                reason=reason_api_text,
+                                assignment_id=result["assignment_id"],
+                                reason_api_text=reason_api_text,
                             )
                             reasons.append(reason)
-                    elif "grade" not in result:
-                        reason_api_text = self.i18n_service.gettext(
-                            result["reason"]
-                            )
-                        reason = self.i18n_service.gettext(
-                            "Assignment {assignment_id}: - {reason_api_text} "
-                        ).format(
-                            assignment_id=result["assignment_id"],
-                            reason_api_text=reason_api_text,
-                        )
-                        reasons.append(reason)
+                else:
+                    LOGGER.exception(
```

Comment:
`exception` adds this to sentry, while `warning` will add it only to logs.

I think we don't need `exception` here since this is a customer error and not something a software engineer can fix.